### ISSUE TITLE
GenCodec for Seq creates a Vector instead of List when reading

### ISSUE THREAD

**Issue by ghik** [AVSystem/scala-commons#76]
Originally reported in #62 

Technically, this is not a bug - `GenCodec.read[Seq[A]]` is free to choose whatever implementation of `Seq` it wants. However, this is non-intuitive because the standard `Seq(...)` factory creates a `List` by default.

It seems that the problem stems from an inconsistency in Scala collections library. `GenCodec.seqCodec[Seq,A]` is based on implicit instance of `CanBuildFrom[Nothing,A,Seq[A]]` which, for whatever reason, creates a `Vector` `¯\_(ツ)_/¯`. This quirk can manifest in simpler examples, e.g.

```scala
scala> Seq(1,2,3)
res19: Seq[Int] = List(1, 2, 3)

scala> Seq(1,2,3).to[Seq]
res18: Seq[Int] = Vector(1, 2, 3)
```

So I'm not sure what to do about it yet, because it's not clear to me what would be the "correct" behaviour here. If we patched `GenCodec` specifically for `Seq` then we'd have to look for all other possible similar inconsistencies in the collections library and patch them accordingly...

**Comment by plokhotnyuk** [comment]
Definitely it would be better to patch `CanBuildFrom[Nothing,A,Seq[A]]` at the Scala repository.

But since 2.13.0-M4 version that trait was replaced by `BuildFrom` and that it seems fixed this inconsistency:
```
andriy@notebook:~$ /opt/scala-2.13.0-M4/bin/scala
Welcome to Scala 2.13.0-M4 (Java HotSpot(TM) 64-Bit Server VM, Java 1.8.0_171).
Type in expressions for evaluation. Or try :help.

scala> Seq(1, 2, 3)
res0: Seq[Int] = List(1, 2, 3)

scala> Seq(1, 2, 3).to(Seq)
res1: Seq[Int] = List(1, 2, 3)
```

**Comment by ghik** [comment]
Fixed in 2.0.0-M1release for Scala 2.13. Won't fix for Scala 2.12

**Comment by ghik** [comment]
Ok, after a second thought it's ugly to have different behaviour in Scala 2.12 and 2.13. Fixed in https://github.com/AVSystem/scala-commons/commit/cfd16b5cc3b084589cbc50802dbb17446f9c1ae1
